xds/src/main/java/io/grpc/xds/XdsServerWrapper.java

```diff
@@ -383,7 +387,21 @@ public void onChanged(final LdsUpdate update) {
         return;
       }
       logger.log(Level.FINEST, "Received Lds update {0}", update);
-      checkNotNull(update.listener(), "update");
+      if (update.listener() == null) {
+        onResourceDoesNotExist("Non-API");
+        return;
+      }
+
+      String ldsAddress = update.listener().address();
+      if (ldsAddress != null && update.listener().protocol() == Protocol.TCP
```

Comment:
From gRFC A36:
> To be useful, the xDS-returned Listener must have an [address](https://github.com/envoyproxy/envoy/blob/11dee4e9c37c244f9de4bc339fdc8695e5de2c5d/api/envoy/config/listener/v3/listener.proto#L108) that matches the listening address provided. The Listener's address would be a TCP SocketAddress with matching address and port_value. The XdsServer must be "not serving" if the address does not match.

If `ldsAddress == null`, then there was not a (supported) address and it should not be serving. If the protocol is not TCP then it should not be serving.

Reply:
This means that every place in our tests that calls `Listener.newBuilder()` needs to set the address.